client/components/credentials-form/index.jsx

```diff
@@ -0,0 +1,130 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { noop } from 'lodash';
+import { localize } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import FormLabel from 'components/forms/form-label';
+import FormPasswordInput from 'components/forms/form-password-input';
+import FormTextInput from 'components/forms/form-text-input';
+import FormButton from 'components/forms/form-button';
+import LoggedOutForm from 'components/logged-out-form';
+import LoggedOutFormFooter from 'components/logged-out-form/footer';
+
+class CredentialsForm extends Component {
+	static propTypes = {
+		submitButtonText: PropTypes.string.isRequired,
+	};
+
+	state = {
+		notice: null,
+		submitting: false,
+		form: null,
+		signedUp: false,
+		validationInitialized: false,
+	};
+
+	getInitialFields() {
+		return {
+			username: '',
+			password: '',
+			submitting: false,
+		};
+	}
+
+	sanitizeUsername( username ) {
+		return username && username.replace( /[^a-zA-Z0-9]/g, '' ).toLowerCase();
```

Comment:
Does this match WordPress username valid chars? Simpler just to pass whatever we are given to the endpoint and show error if we can't login?